src/rules/fileNameCasingRule.ts

```diff
@@ -28,6 +28,74 @@ enum Casing {
     SnakeCase = "snake-case",
 }
 
+type RegexConfig = Record<string, Casing>;
+
+type SimpleConfig = Casing;
+
+type Config = SimpleConfig | RegexConfig;
+
+interface ValidatorSuccessfulResult {
+    valid: true;
+}
+
+interface ValidatorFailureResult {
+    valid: false;
+    failedCasing: Casing;
+}
+
+type ValidatorResult = ValidatorSuccessfulResult | ValidatorFailureResult;
```

Comment:
Type unions are nice but you could get the same behavior with `Casing | undefined` on the validators. If a `Casing` is returned, it would indicate a failure.

Reply:
🤷‍♂️ Hopefully we do not need to return anything else from inside the validator in the future.